drake/util/drakeGeometryUtil.cpp

```diff
@@ -19,168 +19,6 @@ double angleDiff(double phi1, double phi2)
   return d;
 }
 
-
-Vector4d quatConjugate(const Eigen::Vector4d& q)
```

Comment:
while it's nice to see all of this go, it does mean more in the headers, etc.  did you evaluate performance?
one of the big things about all of these geometry utils is that they work on fixed-size matrices (and can potentially avoid most dynamic allocation).  Is the same true for your intializeAutodDiffArgs version?  (I think so, but again it's hard to tell... if yes, please include that in the method documentation)